src/device-manager/java/src/nl/Weave/DataManagement/GenericTraitUpdatableDataSinkImpl.java

```diff
@@ -0,0 +1,444 @@
+    {
+        if (mTraitInstancePtr == 0)
+        {
+            Log.e(TAG, "unexpected err, mTraitInstancePtr is 0");
+            return;
+        }
+        setString(mTraitInstancePtr, path, value, isConditional);
+    }
+
+    @Override
+    public void set(String path, byte[] value, boolean isConditional)
+    {
+        if (mTraitInstancePtr == 0)
+        {
+            Log.e(TAG, "unexpected err, mTraitInstancePtr is 0");
+            return;
+        }
+        setBytes(mTraitInstancePtr, path, value, isConditional);
+    }
+
+    @Override
+    public void setNULL(String path, boolean isConditional)
+    {
+        if (mTraitInstancePtr == 0)
+        {
+            Log.e(TAG, "unexpected err, mTraitInstancePtr is 0");
+            return;
+        }
+        setNULL(mTraitInstancePtr, path, isConditional);
+    }
+
+    @Override
+    public void setSigned(String path, int value)
+    {
+        setSigned(path, value, false);
+    }
+
+    @Override
+    public void setSigned(String path, long value)
+    {
+        setSigned(path, value, false);
+    }
+
+    @Override
+    public void setSigned(String path, BigInteger value)
+    {
+        setSigned(path, value, false);
+    }
+
+    @Override
+    public void setUnsigned(String path, int value)
+    {
+        setUnsigned(path, value, false);
+    }
+
+    @Override
+    public void setUnsigned(String path, long value)
+    {
+        setUnsigned(path, value, false);
+    }
+
+    @Override
+    public void setUnsigned(String path, BigInteger value)
+    {
+        setUnsigned(path, value, false);
+    }
+
+    @Override
+    public void set(String path, double value)
+    {
+        set(path, value, false);
+    }
+
+    @Override
+    public void set(String path, boolean value)
+    {
+        set(path, value, false);
+    }
+
+    @Override
+    public void set(String path, String value)
+    {
+        set(path, value, false);
+    }
+
+    @Override
+    public void set(String path, byte[] value)
+    {
+        set(path, value, false);
+    }
+
+    @Override
+    public void setNULL(String path)
+    {
+        setNULL(path, false);
+    }
+
+    @Override
+    public int getInt(String path)
+    {
+        if (mTraitInstancePtr == 0)
+        {
+            Log.e(TAG, "unexpected err, mTraitInstancePtr is 0");
+            return 0;
+        }
+        return (int)getInt(mTraitInstancePtr, path);
+    }
+
+    @Override
+    public long getLong(String path)
+    {
+        if (mTraitInstancePtr == 0)
+        {
+            Log.e(TAG, "unexpected err, mTraitInstancePtr is 0");
+            return 0;
+        }
+        return (long)getInt(mTraitInstancePtr, path);
+    }
+
+    @Override
+    public BigInteger getBigInteger(String path, int bitLen)
+    {
+        long value = (long)getInt(mTraitInstancePtr, path);
+        String valStr = Long.toString(value);
+        BigInteger bigInt = new BigInteger(valStr);
+        if (mTraitInstancePtr == 0)
```

Comment:
This check should be first, since the `mTraitInstancePtr` is used on line 278

Reply:
Done, thanks